hiku/classes/node.py

```diff
@@ -0,0 +1,160 @@
+import dataclasses as dc
+import typing
+from collections.abc import Hashable
+from functools import partial
+
+import hiku.types
+from hiku.classes.strings import to_snake_case
+from hiku.directives import SchemaDirective
+
+"""
+@node(...)
+class Human:
+    id: int = field(...)
+    droid: ref[Droid] = field_link(...)
+"""
+
+_T = typing.TypeVar("_T", bound=Hashable)
+
+
+class NodeProto(typing.Protocol[_T]):
+    __key__: _T
+
+
+_TNode = typing.TypeVar("_TNode", bound=NodeProto)
+
+
+@dc.dataclass
+class HikuNode:
+    name: str
+    fields: "list[_HikuField] | list[_HikuFieldLink] | list[_HikuField | _HikuFieldLink]"
```

Comment:
This will probably wont work in python<3.10

Reply:
But we need to ensure that this new functionality will work in both old and new python versions. Maybe we can split tests into py39 and py310 using pytest marks and make codebase compatible with all supported python versions ?

Or, just drop support for python < 3.10 )

Reply:
yeah i know, for now i tried to build a simple working prototype, i do plan to make everything backwards-compatible before merging